Tools/TestWebKitAPI/Tests/WebKitCocoa/ServiceWorkerBasic.mm

```diff
@@ -3415,48 +3415,46 @@ HTTPServer server([&] (Connection connection) {
 onload = async () => {
     try {
         // Create cache1
-        cache1 = await self.caches.open("test1");
-        await cache1.put(new Request('test'), new Response('my response'));
+        cache1 = await self.caches.open("cache1");
+        await cache1.put(new Request('my request'), new Response('my response'));
         webkit.messageHandlers.sw.postMessage("PASS");
     } catch(e) {
-        webkit.messageHandlers.sw.postMessage("fill failed with " + e);
+        webkit.messageHandlers.sw.postMessage("put failed with " + e);
     }
 }
 
-function check()
+// The web process may not know that the network process has been terminated since
+// this notification happens asynchronously. So we make calls to open a cache. Until the
+// web process is notified, these calls will fail. Once a call succeeds, we know that the
+// web process has been notified and has established a new connection to a new network
+// process. Then, when match is called, we can check that this new process does not
+// contain cache1, which was created using the now-terminated network process.
+
+async function check()
 {
-    // Create several caches, one of which should have the same ObjectIdentifier value as cache1.
-    Promise.all([
-        self.caches.open("test2"),
-        self.caches.open("test3"),
-        self.caches.open("test4"),
-        self.caches.open("test5"),
-        self.caches.open("test6"),
-        self.caches.open("test7"),
-        self.caches.open("test8"),
-        self.caches.open("test9"),
-        self.caches.open("test10")
-    ]).then(() => {
-        cache1.match('test').then(() => {
-            webkit.messageHandlers.sw.postMessage("cache1 did match unexpectedly");
-        }, e => {
-            webkit.messageHandlers.sw.postMessage("PASS");
-        });
-    }, e => {
-        webkit.messageHandlers.sw.postMessage("check failed with " + e);
-    });
+    var nextCacheNumber = 2;
```

Comment:
Even though the current implementation does not care much about the number of caches, can we increase it to 10 like done previously?

Reply:
Hello! I'm not quite sure I understand. In this version of the test, there's no set number of caches--we just keep attempting to open them until we succeed in opening one. Do you mean set a limit on the number to be 10 as Sihui described? If so, I've done that in the most recent iteration.